nats-base-client/service.ts

```diff
@@ -253,26 +267,28 @@ export type Endpoint = {
   subject: string;
   /**
    * Handler for the endpoint - if not set the service is an iterator
+   * If null, no subscription will be created for the endpoint and instead
```

Comment:
If I recall correctly, we decided that we won't initially add support for making endpoint out of root subject?

It's not a implementation problem, just a cognitive load for users of "if this then this, if not then that" I would like to avoid.

Reply:
So you think it should either be a single root endpoint or multi - but then on then you need to have two factories/constructors, and you break existing ones. The types for the main endpoint is required, so I guess it would be optional, but that doesn't provide much compile time insight.

Reply:
For simplicity and compile safety - Never have "root" endpoint at all?

So to have an endpoint, user have to add it.
That would be not that weird.
You can run http server or whatever without endpoints after all.

Reply:
I agree with @Jarema. I think it will be easier and more straightforward if we simply disallow having handlers for root subject (so it would just serve as a common prefix for a service).

Reply:
@piotrpio after discussion with @ripienaar and @Jarema we don't need to disallow it. This is the simplest case, the thing is that this endpoint is optional. The subject of this endpoint doesn't have anything to do with subject grouping. It is the same as if you add an endpoint listening to a root subject.

Reply:
I will fall in line and say that now that we have addEndpoint() and addGroup() we don't need to have the endpoint. Have made the changes for that and the code looks nice...

Reply:
https://github.com/nats-io/nats-architecture-and-design/pull/186#discussion_r1063024372

Reply:
☝️ is what the code/stats look like